src/snowcli/cli/stage/diff.py

```diff
@@ -137,3 +141,52 @@ def stage_diff(local_path: Path, stage_fqn: str) -> DiffResult:
         result.only_on_stage.append(relpath)
 
     return result
+
+
+def delete_only_on_stage_files(
+    stage_manager: StageManager,
+    stage_fqn: str,
+    only_on_stage: List[str],
+    role: Optional[str] = None,
+):
```

Comment:
Suggest that these three methods go in another file in this same folder, e.g. `stage/operations.py` or something.

Reply:
These are specific to our diff method only, and I'm not sure they will be used beyond the scope of diff. As such, I'll like to keep these two here within diff.

Reply:
Sure, if you feel strongly about it! My counter-argument: I wouldn't expect there to be methods that actually modify anything in a file called `diff`.

Reply:
The other option is `manager.py` whose methods modify a stage in snowflake. I could put these there if the snowCLI team is open to it, but I'd be hesitant to create a different `stage/operations` for it.

Reply:
Definitely up to you! I don't feel extremely strongly.